### PR TITLE
Collect the timings for tasks and groups

### PR DESCRIPTION
**Pull request by klahnakoski**
Capture the task and group timings, push to bigquery

### CODE REVIEW

File: scripts/etl/schedulers/main.py [hunk 1]
```diff
@@ -118,7 +120,61 @@ def version(self, package):
...
+                    {"name":"id", "value":"task.id"},
+                    {"name": "label", "value": "run.name"},
+                    {"name": "start_time", "value": "task.run.start_time"},
+                    {"name": "end_time", "value": "task.run.end_time"},
+                ],
+                "where": [{"eq": {"repo.push.id": push.id}}, {"eq": {"repo.branch.name": push.branch}}],
+                "format": "list",
+                "limit": 100000
+            }
+        )
```
Comment (klahnakoski) [inline scripts/etl/schedulers/main.py]: get all the task timings

File: scripts/etl/schedulers/main.py [hunk 2]
```diff
@@ -118,7 +120,61 @@ def version(self, package):
...
+
+        tasks = {task.id: task for task in task_timings.data}
+
+        for _, tasks_ in jx.chunk((t for t in push.tasks if t.label.startswith("test-")), 10):
```
Comment (klahnakoski) [inline scripts/etl/schedulers/main.py]: for all the tests, group into chunks of 10

File: scripts/etl/schedulers/main.py [hunk 3]
```diff
@@ -118,7 +120,61 @@ def version(self, package):
...
+                    "groupby": [
+                        {"name":"task_id", "value":"task.id"},
+                        {"name": "name", "value": "result.group"}
+                    ],
+                    "where": [{"in": {"task.id": tasks_.id}}],
+                    "format": "list",
+                    "limit": 10000
+                }
+            )
+            for g, groups in jx.groupby(chunk_of_tasks_timings.data, "task_id"):
```
Comment (klahnakoski) [inline scripts/etl/schedulers/main.py:169]: we have 10 tasks, group by task id, leaving a remainder of groups

File: scripts/etl/schedulers/main.py [hunk 4]
```diff
@@ -118,7 +120,61 @@ def version(self, package):
...
+                task = tasks[g.task_id]
+                task.groups = groups
+                task.groups.task_id = None
```
Comment (klahnakoski) [inline scripts/etl/schedulers/main.py:172]: since every group in this task has the same task_id, remove it

**Review by klahnakoski — COMMENTED** [review]
@ahal, @marco-c  getting group timing by task is too slow.  Getting all group timing by push is (sometimes) too big.  May solution was to chunk the tasks and request them that way.

I am not sure how to integrate this with mozci.

The chunking should be done on the ActiveData side; get all tasks/group start and end times in a single request; sent to URL.

**Comment by klahnakoski** [comment]
@marco-c I am asking for you review to look this over at a conceptual level: Only files at `scripts/etl/schedulers` matter, the rest of the changes are library updates

**Comment by marco-c** [comment]
> @ahal, @marco-c getting group timing by task is too slow. Getting all group timing by push is (sometimes) too big. May solution was to chunk the tasks and request them that way.

> I am not sure how to integrate this with mozci.

> The chunking should be done on the ActiveData side; get all tasks/group start and end times in a single request; sent to URL.

You could use the feature built by @worldomonation in https://github.com/mozilla/mozci/pull/229. E.g.:

```Python
summaries = push.make_summary_objects("2020-01-01", "2020-02-01", "autoland", "group")
for summary in summaries:
    group_name = summary.name
    group_duration = summary.median_duration
```

**Pull request merged**